File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/GraphDefinition.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import UsageContext
from base import ContactDetail
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import BackboneElement

@dataclass(kw_only=True)
class GraphDefinitionLinkTargetCompartment(BackboneElement):
    use: str
    code: str
    rule: str
    expression: Optional[str] = None
    description: Optional[str] = None

@dataclass(kw_only=True)
class GraphDefinitionLinkTarget(BackboneElement):
    link: Optional[List[Reference]] = field(default_factory=list)
    type: str
    params: Optional[str] = None
    profile: Optional[str] = None
    compartment: Optional[List[GraphDefinitionLinkTargetCompartment]] = field(default_factory=list)

@dataclass(kw_only=True)
class GraphDefinitionLink(BackboneElement):
    max: Optional[str] = None
    min: Optional[int] = None
    path: Optional[str] = None
    target: Optional[List[GraphDefinitionLinkTarget]] = field(default_factory=list)
    slice_name: Optional[str] = None
    description: Optional[str] = None

@dataclass(kw_only=True)
class GraphDefinition(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    publisher: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = field(default_factory=list)
    purpose: Optional[str] = None
    name: str
    start: str
    use_context: Optional[List[UsageContext]] = field(default_factory=list)
    experimental: Optional[bool] = None
    status: str
    link: Optional[List[GraphDefinitionLink]] = field(default_factory=list)
    url: Optional[str] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = field(default_factory=list)
    profile: Optional[str] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/GraphDefinition.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class GraphDefinition : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? Date { get; set; }
    public string? Publisher { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public string? Purpose { get; set; }
    public required string Name { get; set; }
    public required string Start { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public bool? Experimental { get; set; }
    public required string Status { get; set; }
    public GraphDefinition_Link[]? Link { get; set; }
    public string? Url { get; set; }
    public string? Version { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }
    public string? Profile { get; set; }

    public class GraphDefinition_Link_Target_Link_Target_Compartment : BackboneElement
    {
        public required string Use { get; set; }
        public required string Code { get; set; }
        public required string Rule { get; set; }
        public string? Expression { get; set; }
        public string? Description { get; set; }
    }

    public class GraphDefinition_Link_Target_Link_Target : BackboneElement
    {
        public string[]? Link { get; set; }
        public required string Type { get; set; }
        public string? Params { get; set; }
        public string? Profile { get; set; }
        public GraphDefinition_Link_Target_Link_Target_Compartment[]? Compartment { get; set; }
    }

    public class GraphDefinition_Link_Target_Link : BackboneElement
    {
        public string? Max { get; set; }
        public int? Min { get; set; }
        public string? Path { get; set; }
        public GraphDefinition_Link_Target_Link_Target[]? Target { get; set; }
        public string? SliceName { get; set; }
        public string? Description { get; set; }
    }

    public class GraphDefinition_Link_Target_Compartment : BackboneElement
    {
        public required string Use { get; set; }
        public required string Code { get; set; }
        public required string Rule { get; set; }
        public string? Expression { get; set; }
        public string? Description { get; set; }
    }

    public class GraphDefinition_Link_Target : BackboneElement
    {
        public GraphDefinition_Link_Target_Link[]? Link { get; set; }
        public required string Type { get; set; }
        public string? Params { get; set; }
        public string? Profile { get; set; }
        public GraphDefinition_Link_Target_Compartment[]? Compartment { get; set; }
    }

    public class GraphDefinition_Link : BackboneElement
    {
        public string? Max { get; set; }
        public int? Min { get; set; }
        public string? Path { get; set; }
        public GraphDefinition_Link_Target[]? Target { get; set; }
        public string? SliceName { get; set; }
        public string? Description { get; set; }
    }
}